api/v1beta1/region_types.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2025
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	fluxmeta "github.com/fluxcd/pkg/apis/meta"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	RegionKind = "Region"
+
+	RegionFinalizer = "k0rdent.mirantis.com/region"
+
+	KCMRegionLabelKey = "k0rdent.mirantis.com/region"
+)
+
+// +kubebuilder:validation:XValidation:rule="has(self.kubeConfig) != has(self.clusterDeployment)",message="exactly one of kubeConfig or clusterDeployment must be set"
+
+// RegionSpec defines the desired state of Region
+type RegionSpec struct {
+	// KubeConfig references the Secret containing the kubeconfig
+	// of the cluster being onboarded as a regional cluster.
+	// The Secret must reside in the system namespace.
+	KubeConfig *fluxmeta.SecretKeyReference `json:"kubeConfig,omitempty"`
+
+	// ClusterDeployment is the reference to the existing ClusterDeployment object
+	// to be onboarded as a regional cluster.
+	ClusterDeployment *corev1.ObjectReference `json:"clusterDeployment,omitempty"`
```

Comment:
isn't it better to either create our own structure for reference of `cld` or simplify with just a namespaced name? i believe the latter should suffice